src/System/Hworker.hs

```diff
@@ -434,6 +622,48 @@ debugger microseconds hw =
                         (\queued -> hwlog hw ("DEBUG", queued, running)))
      threadDelay microseconds
 
+-- | Initializes a batch of jobs. By default the information for tracking a
+-- batch of jobs, created by this function, will expires a week from
+-- its creation. The optional `seconds` argument can be used to override this.
+initBatch :: Hworker s t -> Maybe Integer -> IO BatchId
+initBatch hw seconds = do
+  batch <- BatchId <$> UUID.nextRandom
+  void . R.runRedis (hworkerConnection hw) $ do
+    _ <-
+      R.hmset (batchCounter hw batch)
+        [ ("queued", "0")
+        , ("completed", "0")
+        , ("successes", "0")
+        , ("failures", "0")
+        , ("retries", "0")
+        , ("status", "queueing")
+        ]
+    R.expire (batchCounter hw batch) (fromMaybe 604800 seconds)
+  return batch
```

Comment:
I guess it's possible that the Redis command could fail. I wonder if this should actually return something like `IO (Either Text BatchId)` in the case that the counter doesn't get set up?

Reply:
Also, I wonder if there should be an option to not expire the batch summary at all?

Reply:
Seems worth checking that the hmset works... as if it ever doesn't, things are going to be very messed up :) 

As to never expiring -- given that the summary doesn't take up much space, keeping things around doesn't seem unreasonable, so you could make it so that if Nothing was passed it never expired...

Reply:
Makes sense. Does `IO (Either Text BatchId)` seem ok as a return value? Or should it actually be throwing an error or something?